code/app/src/main/java/ca/ualberta/ishelf/myBooksFragment.java

```diff
@@ -258,7 +260,7 @@ public void onDataChange(DataSnapshot dataSnapshot) {
                         Gson gson = new Gson();
                         Type tokenType = new TypeToken<Book>() {}.getType();
                         Book book = gson.fromJson(jBook, tokenType); // here is where we get the user object
-                        if (ownedBooksIds.contains(book.getId())){
+                        if (ownedBooksIds.contains(book.getId()) || book.getOwner().equals(currentUsername)){
```

Comment:
Is it || or &&

Reply:
I made it or because the syncing problem in #129 was that the book is added to firebase, but adding it to the owner list of books was still happening in another thread and not completed, so the previous line (in red) didn't add the book bc its still not in the owner list of books, but now it also checks if you're the owner and adds it anyway.